src/common/Utilities/Util.cpp

```diff
@@ -297,34 +304,40 @@ void utf8truncate(std::string& utf8str, size_t len)
         char* oend = utf8::utf16to8(wstr.c_str(), wstr.c_str()+wstr.size(), &utf8str[0]);
         utf8str.resize(oend-(&utf8str[0]));                 // remove unused tail
     }
-    catch(std::exception)
+    catch(std::exception const&)
     {
-        utf8str = "";
+        utf8str.clear();
     }
 }
 
 bool Utf8toWStr(char const* utf8str, size_t csize, wchar_t* wstr, size_t& wsize)
 {
     try
     {
-        size_t len = utf8::distance(utf8str, utf8str+csize);
-        if (len > wsize)
+        Trinity::CheckedBufferOutputIterator<wchar_t> out(wstr, wsize);
+        out = utf8::utf8to16(utf8str, utf8str+csize, out);
+        wsize -= out.remaining(); // remaining unused space
+        wstr[wsize] = L'\0';
+    }
+    catch(std::exception const&)
+    {
+        // Replace the converted string with an error message if there is enough space
+        // Otherwise just return an empty string
+        wchar_t const* errorMessage = L"An error occurred converting string from UTF-8 to WStr";
+        size_t errorMessageLength = wcslen(errorMessage);
+        if (wsize >= errorMessageLength)
+        {
+            wcscpy(wstr, errorMessage);
+            wsize = wcslen(wstr);
+        }
```

Comment:
Perhaps this could fix the Codacy errors:
```cpp
        wchar_t errorMessage[] = L"An error occurred converting string from UTF-8 to WStr";
        size_t errorMessageLength = (sizeof errorMessage / sizeof *errorMessage) - 1; // size without null character
        if (wsize >= errorMessageLength)
        {
            wcscpy(wstr, errorMessage);
            wsize = errorMessageLength;
        }
```

Reply:
its non-sense codacy warrning they just don't analyze further code

Reply:
we don't need to suck codacy balls here we can just ignore that

Reply:
its also can cause a crash in some win32 cases :/

Reply:
done by more proper way